File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>
using namespace std;

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    vector<bool> vis(n, false);
    
    for(int i=0; i<n; i++) {
        if(vis[i] == true) continue;
        answer++;
        queue<int> q;
        q.push(i);
        vis[i] = true;
        
        while(!q.empty()) {
            int cur = q.front(); q.pop();
            for(int j=0; j<n; j++) {
                if(computers[cur][j] == 1 && vis[j] == false) {
                    vis[j] = true;
                    q.push(j);
                }
            }
        }
    }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    static boolean[] vis;
    static int cnt;
    public int solution(int n, int[][] computers) {
        cnt = 0;
        vis = new boolean[n];
        for(int i=0; i<n; i++) {
            if(!vis[i]) {
                dfs(computers, i);
                cnt++;
            }
        }        
        return cnt;
    }
    static void dfs(int[][] computers, int node) {
        vis[node] = true;
        for(int i=0; i<computers.length; i++) {
            if(computers[node][i] == 1 && !vis[i]) {
                dfs(computers, i);
            }
        }
    }
}